new file:   allure_results/0e9e8ca6-0efc-48a9-8a26-d24938927197-resu…

NastyaPilat opened the pull request:
…lt.json

	new file:   allure_results/155fc26c-2c9f-4d8f-80af-44a80e03710e-container.json
	new file:   allure_results/206febaf-319f-4126-80ef-595a52588304-result.json
	new file:   allure_results/213c85cd-1d07-4ba8-b3c8-e276a1e21a79-result.json
	new file:   allure_results/2b7225a3-b76c-44ce-aeaa-851eb1b608ed-result.json
	new file:   allure_results/2d8151e5-b87a-4a1c-95e6-44f52bf3bad8-result.json
	new file:   allure_results/330fed5f-b3fd-43d7-ae25-64f0a36ddca0-result.json
	new file:   allure_results/3fb6908a-fa06-4537-acaa-d0309b030c2c-result.json
	new file:   allure_results/667fdb01-6c1e-4b55-a49d-f8973aa0619b-container.json
	new file:   allure_results/9cf7283d-be98-4ca1-a50c-f2cde90eb687-result.json
	new file:   allure_results/9f9b7fb4-847a-4e45-98ce-a25ac57e813f-result.json
	new file:   allure_results/a9adc728-f32f-4456-90b8-ae43583da514-result.json
	new file:   allure_results/abda077c-7368-446e-81b5-b20708cd766f-container.json
	new file:   allure_results/adf79859-2ae4-4175-a075-f89bb800f4cf-result.json
	new file:   allure_results/c3070f69-5e22-4033-a385-a0705dc64d40-result.json
	new file:   allure_results/ccd8ea67-a15a-437c-bf2f-6c8b675bf76f-result.json
	new file:   allure_results/dc407dfe-cb53-4ceb-8811-0ca5cf9fbeed-result.json

maxpilat committed:
new file:   allure_results/0e9e8ca6-0efc-48a9-8a26-d24938927197-result.json
	new file:   allure_results/155fc26c-2c9f-4d8f-80af-44a80e03710e-container.json
	new file:   allure_results/206febaf-319f-4126-80ef-595a52588304-result.json
	new file:   allure_results/213c85cd-1d07-4ba8-b3c8-e276a1e21a79-result.json
	new file:   allure_results/2b7225a3-b76c-44ce-aeaa-851eb1b608ed-result.json
	new file:   allure_results/2d8151e5-b87a-4a1c-95e6-44f52bf3bad8-result.json
	new file:   allure_results/330fed5f-b3fd-43d7-ae25-64f0a36ddca0-result.json
	new file:   allure_results/3fb6908a-fa06-4537-acaa-d0309b030c2c-result.json
	new file:   allure_results/667fdb01-6c1e-4b55-a49d-f8973aa0619b-container.json
	new file:   allure_results/9cf7283d-be98-4ca1-a50c-f2cde90eb687-result.json
	new file:   allure_results/9f9b7fb4-847a-4e45-98ce-a25ac57e813f-result.json
	new file:   allure_results/a9adc728-f32f-4456-90b8-ae43583da514-result.json
	new file:   allure_results/abda077c-7368-446e-81b5-b20708cd766f-container.json
	new file:   allure_results/adf79859-2ae4-4175-a075-f89bb800f4cf-result.json
	new file:   allure_results/c3070f69-5e22-4033-a385-a0705dc64d40-result.json
	new file:   allure_results/ccd8ea67-a15a-437c-bf2f-6c8b675bf76f-result.json
	new file:   allure_results/dc407dfe-cb53-4ceb-8811-0ca5cf9fbeed-result.json

maxpilat committed:
new file:   .gitignore
	new file:   constants.py
	new file:   requirements.txt
	new file:   tests/__init__.py
	new file:   tests/test_create_courier.py
	new file:   tests/test_create_order.py
	new file:   tests/test_login_courier.py
	new file:   tests/test_order_list.py
	new file:   utils.py

AlexKitIt commented on tests/test_create_courier.py:
```diff
@@ -0,0 +1,40 @@
+    @allure.title('Нельзя создать двух одинаковых курьеров')
+    def test_create_duplicate_courier(self):
+        courier_data = register_new_courier_and_return_login_password()[0]
+        response = requests.post(constants.BASE_URL + "/api/v1/courier", data={
```
Можно лучше: эндпоинты тоже стоит хранить отдельно, они постоянно переиспользуются

AlexKitIt commented on tests/test_create_courier.py:
```diff
@@ -0,0 +1,40 @@
+            "password": courier_data[1],
+            "firstName": courier_data[2]
+        })
+        assert response.status_code == 409
```
Нужно исправить: здесь и далее: одной проверки на статус код недостаточно, необходимо парсить ответ

AlexKitIt commented on tests/test_create_courier.py:
```diff
@@ -0,0 +1,40 @@
+        })
+        assert response.status_code == 400
+
+    @allure.title('Запрос возвращает правильный код ответа')
+    def test_create_courier_returns_correct_status_code(self):
+        response = register_new_courier_and_return_login_password()[1]
+        assert response.status_code == 201
+
+    @allure.title('Успешный запрос возвращает ok: true')
+    def test_create_courier_returns_ok_on_success(self):
+        response = register_new_courier_and_return_login_password()[1]
+        assert response.json()['ok']
```
Нужно исправить: это один тест

AlexKitIt commented on utils.py:
```diff
@@ -0,0 +1,43 @@
+# если регистрация не удалась, возвращает пустой список
+
+
+def register_new_courier_and_return_login_password():
```
Нужно исправить: над каждым логиеческим методом ставится аннотация allure.step

maxpilat committed:
new file:   allure_results/0a5bcbce-bc96-4156-9b03-d0c103ddc4d5-container.json
	deleted:    allure_results/0e9e8ca6-0efc-48a9-8a26-d24938927197-result.json
	deleted:    allure_results/155fc26c-2c9f-4d8f-80af-44a80e03710e-container.json
	deleted:    allure_results/206febaf-319f-4126-80ef-595a52588304-result.json
	deleted:    allure_results/213c85cd-1d07-4ba8-b3c8-e276a1e21a79-result.json
	deleted:    allure_results/2b7225a3-b76c-44ce-aeaa-851eb1b608ed-result.json
	deleted:    allure_results/2d8151e5-b87a-4a1c-95e6-44f52bf3bad8-result.json
	new file:   allure_results/2ec7cc3b-265e-4cb7-9cce-3bd4b06042f4-result.json
	new file:   allure_results/4b4dce71-0216-4b68-9dd2-c51448714c51-container.json
	renamed:    allure_results/ccd8ea67-a15a-437c-bf2f-6c8b675bf76f-result.json -> allure_results/4bb945fd-4b99-4d5d-bbd2-a8938df2f148-result.json
	new file:   allure_results/603842c8-03a4-45b4-9b1f-0f756c528fcb-result.json
	deleted:    allure_results/667fdb01-6c1e-4b55-a49d-f8973aa0619b-container.json
	new file:   allure_results/7189f0d0-9ea7-4773-864d-24b604f5fb15-result.json
	renamed:    allure_results/330fed5f-b3fd-43d7-ae25-64f0a36ddca0-result.json -> allure_results/733865a1-aff7-457d-963f-a5ef2a3305eb-result.json
	new file:   allure_results/745b9fdd-f86d-4ef5-8658-b52c69a0dd0b-result.json
	new file:   allure_results/8059434d-a04a-45ce-94d3-4a0261f84c45-container.json
	deleted:    allure_results/9cf7283d-be98-4ca1-a50c-f2cde90eb687-result.json
	new file:   allure_results/a04b981d-1312-4efb-9c9f-ad7b767c77b4-result.json
	renamed:    allure_results/9f9b7fb4-847a-4e45-98ce-a25ac57e813f-result.json -> allure_results/a344b5b1-1f45-49ae-89eb-8c357203bdd9-result.json
	deleted:    allure_results/a9adc728-f32f-4456-90b8-ae43583da514-result.json
	new file:   allure_results/ab2d9435-0df0-4579-9d60-58f002e04726-result.json
	deleted:    allure_results/abda077c-7368-446e-81b5-b20708cd766f-container.json
	deleted:    allure_results/adf79859-2ae4-4175-a075-f89bb800f4cf-result.json
	renamed:    allure_results/dc407dfe-cb53-4ceb-8811-0ca5cf9fbeed-result.json -> allure_results/c280119a-9395-460b-ad72-d9105998c32f-result.json
	deleted:    allure_results/c3070f69-5e22-4033-a385-a0705dc64d40-result.json
	renamed:    allure_results/3fb6908a-fa06-4537-acaa-d0309b030c2c-result.json -> allure_results/cbf8ac27-811c-4e8f-beeb-3a30053a977f-result.json
	modified:   constants.py
	modified:   tests/test_create_courier.py
	modified:   tests/test_create_order.py
	modified:   tests/test_login_courier.py
	modified:   tests/test_order_list.py
	modified:   utils.py

maxpilat committed:
new file:   allure_results/03e4bd62-25c2-4a4f-bfaa-23d18597f8d3-container.json
	deleted:    allure_results/0a5bcbce-bc96-4156-9b03-d0c103ddc4d5-container.json
	renamed:    allure_results/c280119a-9395-460b-ad72-d9105998c32f-result.json -> allure_results/129ee188-d223-4e82-98ac-c730a4236014-result.json
	new file:   allure_results/1f0c91a6-0714-4ee3-955e-6ad2106959ed-container.json
	renamed:    allure_results/733865a1-aff7-457d-963f-a5ef2a3305eb-result.json -> allure_results/3ebb6b13-1d74-4ea2-ac25-1c6357f255f2-result.json
	renamed:    allure_results/cbf8ac27-811c-4e8f-beeb-3a30053a977f-result.json -> allure_results/43fdcf46-eaea-4fdb-924f-cadea66ba40f-result.json
	deleted:    allure_results/4b4dce71-0216-4b68-9dd2-c51448714c51-container.json
	renamed:    allure_results/7189f0d0-9ea7-4773-864d-24b604f5fb15-result.json -> allure_results/53479bba-c523-41b9-bc62-cd81dfd48e18-result.json
	renamed:    allure_results/2ec7cc3b-265e-4cb7-9cce-3bd4b06042f4-result.json -> allure_results/551ac79f-23ff-4659-9526-de44c1aa167f-result.json
	deleted:    allure_results/603842c8-03a4-45b4-9b1f-0f756c528fcb-result.json
	renamed:    allure_results/745b9fdd-f86d-4ef5-8658-b52c69a0dd0b-result.json -> allure_results/66ccc90c-557f-4a3d-b39d-92d1ba051996-result.json
	new file:   allure_results/7e165244-936a-48f8-b6c3-2f32726f5578-container.json
	renamed:    allure_results/4bb945fd-4b99-4d5d-bbd2-a8938df2f148-result.json -> allure_results/7f5068be-dfc0-4184-ab8f-adccdc2917cc-result.json
	renamed:    allure_results/a344b5b1-1f45-49ae-89eb-8c357203bdd9-result.json -> allure_results/7ff07023-3870-4baf-bf2a-2ce9f2ca7a9f-result.json
	deleted:    allure_results/8059434d-a04a-45ce-94d3-4a0261f84c45-container.json
	renamed:    allure_results/ab2d9435-0df0-4579-9d60-58f002e04726-result.json -> allure_results/b5d4163b-ec17-4891-9466-d39b9fe4a2bb-result.json
	renamed:    allure_results/a04b981d-1312-4efb-9c9f-ad7b767c77b4-result.json -> allure_results/fb84236a-bd82-4168-b374-02f01812f0e7-result.json
	modified:   tests/test_create_courier.py

AlexKitIt commented on tests/test_create_courier.py:
```diff
@@ -0,0 +1,32 @@
+            "password": courier_data[1],
+            "firstName": courier_data[2]
+        })
+        assert response.status_code == 409 and response.json(
```
Можно лучше: тексты ответов стоит хранить снаружи в data модулях для поддержания и актуализации
